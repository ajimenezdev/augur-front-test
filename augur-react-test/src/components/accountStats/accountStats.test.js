import React from "react";
import renderer from "react-test-renderer";
import { fireEvent, render, waitForElement } from "@testing-library/react";
import { act } from "@testing-library/react-hooks";
import "@testing-library/jest-dom/extend-expect";
import AccountStats from "./index";

it("renders without crashing", () => {
  const component = renderer.create(
    <AccountStats
      account=""
      accountStats={{}}
      onAccountChange={() => {}}
      onForceRefresh={() => {}}
    />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders with arguments all values properly", () => {
  act(() => {
    const { getByText, getByLabelText } = render(
      <AccountStats
        account="0xfbb1b73c4f0bda4f67dca266ce6ef42f520fbb98"
        accountStats={{
          balance: 1230000000000000000,
          balanceFetching: false,
          transactionsCount: { outgoingTx: 10, incomingTx: 15 },
          transactionsCountFetching: false
        }}
        onAccountChange={() => {}}
        onForceRefresh={() => {}}
      />
    );

    expect(getByText(/0xfbb1b73c4f0bda4f67dca266ce6ef42f520fbb98/i));
    expect(getByLabelText("account-balance")).toHaveTextContent("1.23");
    expect(getByLabelText("account-incoming")).toHaveTextContent("15");
    expect(getByLabelText("account-outgoing")).toHaveTextContent("10");
  });
});

it("execute callback when button is clicked", async () => {
  await act(async () => {
    const mockAccount = jest.fn();
    const mockRefresh = jest.fn();
    const { getByLabelText, container } = render(
      <AccountStats
        account="0xfbb1b73c4f0bda4f67dca266ce6ef42f520fbb98"
        accountStats={{
          balance: 1230000000000000000,
          balanceFetching: false,
          transactionsCount: { outgoingTx: 10, incomingTx: 15 },
          transactionsCountFetching: false
        }}
        onAccountChange={mockAccount}
        onForceRefresh={mockRefresh}
      />
    );

    const buttonRefresh = getByLabelText("refresh-accountStats");
    fireEvent.click(buttonRefresh);

    expect(mockAccount.mock.calls.length).toBe(0);
    expect(mockRefresh.mock.calls.length).toBe(1);

    const buttonEdit = getByLabelText("edit-account");
    fireEvent.click(buttonEdit);

    await waitForElement(() => getByLabelText("address"), {
      container
    });

    fireEvent.change(getByLabelText("address"), {
      target: { value: "0xc350de406f54271c0e025bf72855418d65f8bcbc" }
    });
    const buttonSave = getByLabelText("save-button");
    fireEvent.click(buttonSave);

    expect(mockRefresh.mock.calls.length).toBe(1);
    expect(mockAccount.mock.calls.length).toBe(1);
  });
});
