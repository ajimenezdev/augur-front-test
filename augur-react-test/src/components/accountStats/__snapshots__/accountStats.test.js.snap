// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation1 MuiCard-root accountStats MuiPaper-rounded"
>
  <div
    className="MuiCardHeader-root"
  >
    <div
      className="MuiCardHeader-content"
    >
      <span
        className="MuiTypography-root MuiCardHeader-title MuiTypography-h5 MuiTypography-displayBlock"
      >
        Account Stats
      </span>
      <span
        className="MuiTypography-root MuiCardHeader-subheader MuiTypography-body1 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
      >
        
      </span>
    </div>
    <div
      className="MuiCardHeader-action"
    >
      <button
        aria-label="edit-account"
        className="MuiButtonBase-root MuiIconButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiIconButton-label"
        >
          <span
            aria-hidden={true}
            className="material-icons MuiIcon-root accountStats-refreshIcon"
          >
            edit
          </span>
        </span>
      </button>
      <button
        aria-label="refresh-accountStats"
        className="MuiButtonBase-root MuiIconButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiIconButton-label"
        >
          <span
            aria-hidden={true}
            className="material-icons MuiIcon-root accountStats-refreshIcon"
          >
            refresh
          </span>
        </span>
      </button>
    </div>
  </div>
  <div
    className="MuiCardContent-root accountStats-content"
  >
    <div
      className="accountStats-infoRow"
    >
      <span
        className="accountStats-label"
      >
        Balance:
      </span>
      <span
        aria-label="account-balance"
        className="accountStats-value"
      />
    </div>
    <hr
      className="MuiDivider-root"
    />
    <div
      className="accountStats-infoRow"
    >
      <span
        className="accountStats-label"
      >
        Outgoing Tx:
      </span>
      <span
        aria-label="account-outgoing"
        className="accountStats-value"
      />
    </div>
    <hr
      className="MuiDivider-root"
    />
    <div
      className="accountStats-infoRow"
    >
      <span
        className="accountStats-label"
      >
        Incoming Tx:
      </span>
      <span
        aria-label="account-incoming"
        className="accountStats-value"
      />
    </div>
  </div>
</div>
`;
